### PR TITLE
Adds gladiator cron

### LINKED ISSUE #6225: Create cron to set up competitions
in order to only have one place to configure competitions
phoenix can ping gladiator to get the info and set the proper variables to set up a competition

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?

Adds cron into phoenix which pings gladiator and updates all of the campaigns here on the drupals.
#### How should this be manually tested?
- start phoenix + gladiator
- change GLADIATOR_URL in the the ds_gladiator module if its diff
- run cron 
- are things updated correctly?

heres an example for node 1485 on my local

changed the date so the contest is over
screen shot 2016-03-18 at 10 24 21 am

changed the date so the contest is on
screen shot 2016-03-18 at 10 19 41 am
#### Any background context you want to provide?

no not really
#### What are the relevant tickets?

Fixes #6225

### CODE REVIEW

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     'headers' => $client['headers'],
+   ];
+
+   $response = drupal_http_request($client['base_url'] . '/contests', $options);
+   $contests = json_decode($response->data)->data;
+
+   foreach ($contests as $contest) {
+     $nid = $contest->campaign->id;
+     // For some reason using the boolean value doesnt work in the fields array, have to convert to int.
+     $signupsOpen = $contest->waiting_room->open ? 1 : 0;
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc]: `camel_case_vars_in_drupal` :camel:

**Comment by angaither** [comment]
@mikefantini when we turned on/off the signup prompt today, you had Hilary turn off `collect signup data` does that make the most sense?
In this PR @deadlybutter is toggling `prompt after signup` and `competition signup`

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+     db_update('dosomething_signup_data_form')
+       ->fields([
+         'required' => $signupsOpen,
+         'competition_signup' => $signupsOpen,
+       ])
+       ->condition('nid', $nid)
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc:31]: does this work if the form doesn't exist already?

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ function dosomething_gladiator_cron() {
+   $client = _dosomething_gladiator_build_http_client();
+
+   $options = [
+     'method' => 'GET',
+     'headers' => $client['headers'],
+   ];
+
+   $response = drupal_http_request($client['base_url'] . '/contests', $options);
+   $contests = json_decode($response->data)->data;
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc:20]: should check that the response is valid & has data before using the data

**Comment by mikefantini** [comment]
@angaither @deadlybutter IMO when we no longer want to prompt users to join a competition it makes sense to just disable the entire `collect additional signup data`. Do you see any reason we wouldn't want to do that?

From what I've seen the values in the other fields remain so we would be able to keep copy etc. if we wanted to have a competition again next year.

**Comment by angaither** [comment]
@mikefantini makes sense! @deadlybutter you can just disable the whole thing.

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc (hunk 1 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc]: ahh right right

**Comment by itsjoekent** [comment]
thats what i first tried but ran into problems, so i just disabled the other stuffs. i can try again if its a problem

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc (hunk 2 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc:31]: hmmm, why wouldnt the form exist?

Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc:31]: if a contest is created in gladiator, but the language isn't set up here yet

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc:31]: ah I see, but shouldnt that resolve itself once someone updates the campaign & cron runs again?

**Comment by angaither** [comment]
@deadlybutter what problems when you disabled `collect user address`

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+   $client = _dosomething_gladiator_build_http_client();
+
+   $options = [
+     'method' => 'GET',
+     'headers' => $client['headers'],
+   ];
+
+   $response = drupal_http_request($client['base_url'] . '/contests', $options);
+
+   if ($response->code != 200 ) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc]: :space_invader: extra space after `200`

**Comment by itsjoekent** [comment]
wait thats what its called?

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+     return;
+   }
+
+   $contests = json_decode($response->data)->data;
+
+   foreach ($contests as $contest) {
+     $nid = $contest->campaign->id;
+     // For some reason using the boolean value doesnt work in the fields array, have to convert to int.
+     $signups_open = $contest->waiting_room->open ? 1 : 0;
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc]: small note, but it's more readable to use a statement as the variable name here `signup_is_open` or `contest_is_open`

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc]: sure thing!

**Comment by itsjoekent** [comment]
ill try disabling `collect user address` as well if thats the field name, one sec

**Comment by itsjoekent** [comment]
i was trying something_required originally

File: lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc (hunk 2 above)
Comment (angaither) [inline lib/modules/dosomething/dosomething_gladiator/dosomething_gladiator.cron.inc:31]: I'm not sure that's why I asked if it works if the form doesn't exist.

**Comment by itsjoekent** [comment]
yeah I cant get that bit to work @angaither :/ just seems to fail

**Comment by angaither** [comment]
@deadlybutter what's the field name you're using? does it match the schema?

**Comment by itsjoekent** [comment]
@angaither should be ready for re-review!

**Comment by angaither** [comment]
@deadlybutter did you test this with a node where the data form doesn't exist?

**Comment by itsjoekent** [comment]
cant see my sql db so no I havent, sena is gonna help me figure it out in a bit

**Comment by itsjoekent** [comment]
so if the signup form data does not exist, this cron will not effect it

**Comment by itsjoekent** [comment]
i think thats ok to merge then? @angaither

**Comment by angaither** [comment]
:+1:

**Pull request merged**